src/System.Memory/tests/ReadOnlySpan/Overlaps.cs

```diff
@@ -0,0 +1,972 @@
+        //      second:            [---------)
+        //                        100       200
+        //
+        [InlineData(200, 200, 100, 200, false)]
+
+        //
+        //                                  200
+        //       first:                      |
+        //      second:            [-------------------)
+        //                        100                 300
+        //
+        [InlineData(200, 200, 100, 300, false)]
+
+        //
+        //                                  200
+        //       first:                      |
+        //      second:                      |
+        //                                  200
+        //
+        [InlineData(200, 200, 200, 200, false)]
+
+        //
+        //                                  200
+        //       first:                      |
+        //      second:                      [---------)
+        //                                  200       300
+        //
+        [InlineData(200, 200, 200, 300, false)]
+
+        //
+        //                                  200
+        //       first:                      |
+        //      second:                                |
+        //                                            300
+        //
+        [InlineData(200, 200, 300, 300, false)]
+
+        //
+        //                                  200       300
+        //       first:                      [---------)
+        //      second:  |
+        //               0
+        //
+        [InlineData(200, 300, 0, 0, false)]
+
+        //
+        //                                  200       300
+        //       first:                      [---------)
+        //      second:  [---------)
+        //               0        100
+        //
+        [InlineData(200, 300, 0, 100, false)]
+
+        //
+        //                                  200       300
+        //       first:                      [---------)
+        //      second:  [-------------------)
+        //               0                  200
+        //
+        [InlineData(200, 300, 0, 200, false)]
+
+        //
+        //                                  200       300
+        //       first:                      [---------)
+        //      second:  [-----------------------------)
+        //               0                            300
+        //
+        [InlineData(200, 300, 0, 300, true)]
+
+        //
+        //                                  200       300
+        //       first:                      [---------)
+        //      second:            |
+        //                        100
+        //
+        [InlineData(200, 300, 100, 100, false)]
+
+        //
+        //                                  200       300
+        //       first:                      [---------)
+        //      second:            [---------)
+        //                        100       200
+        //
+        [InlineData(200, 300, 100, 200, false)]
```

Comment:
Is this case different from `[InlineData(100, 200, 0, 100, false)]` above?

Reply:
No. I simply generated *all* combinations that are possible. I didn't spend any time on de-duplicating equivalent test cases.